File: PythonAlgorithmes/Problem-solving/FindLongestWordInAString.py
# First solution

#def longest_word_in(sentence):
#    count = 0
#    word_list = sentence.split("  ")
#    for word in word_list:
#        if len (word) > count:
#            count = len(word)
#            longest = word

#    return longest 

# Second solution

def longest_word_in(sentence):
    longest = ""
    word_list = sentence.split("  ")
    for word in word_list:
        if len (word) > len(longest):
            count = len(word)
            longest = word

    return longest


# Testing Output
# Toooooo
print(longest_word_in("In Programming We Love Elzero Academy Tooooooo Much"))

File: JavascriptAlgorithmes/Problem-solving/FindLongestWordInAString.js
// First solution:

/* function longestWordIn(sentence) {
    let splArray = sentence.split("  ");
    let longestLen = 0;
    let longestWord;


    for (let i = 0; i < splArray.length; i++) {
        if (splArray[1].length > longestLen) {
            longestLen = splArray[i];
            longestLen = splArray[i].length;
        }
    }

    return splArray;
} */

// Second solution using rediuce

function longestWordIn(sentence) {
    let longestWord = sentence.split("  ").reduce(function(longest, current) {
        return current.legth > longest.length ? current : longest;
    })

    return splArray;
}

console.log(longestWordIn("In Programming We Love Elzero Academy Tooooo Much"))